Core/src/main/java/put/io/black/java/core/logic/ScenarioManager.java

```diff
@@ -1,450 +1,493 @@
+    }
+
+    /**
+     * Generate scenario with numerations. New nesting level use parent number and create block 1. 1.1. 1.2 ...
+     * @return Scenario with numerations
+     */
+    public String getScenarioWithNumeration() {
+        LinkedList<String> scenarioWithNumeration = new LinkedList<>();
+        for (int i = 0; i < getFirstLevelNodes().size(); i++) {
+            if (getFirstLevelNodes().get(i) instanceof KeyNode) {
+                if (((KeyNode) getFirstLevelNodes().get(i)).hasChildren()) {
+                    addLineWithNumeration(getFirstLevelNodes().get(i), scenarioWithNumeration, (i + 1) + ".");
+                }
+            } else {
+                String line = makeTabulaturePrefix(getFirstLevelNodes().get(i).getNestingLevel()) + (i + 1) + "." + getFirstLevelNodes().get(i).getLine() + "\n";
+                scenarioWithNumeration.addLast(line);
+            }
+
+        }
+        return changeLinkedListToString(scenarioWithNumeration);
+    }
+
+    /**
+     * Add line with numeration to scenario
+     * @param node Node
+     * @param scenarioWithNumeration Scenario with numeration list
+     * @param prefix Prefix to prepend (number block)
+     */
+    private void addLineWithNumeration(Node node, LinkedList<String> scenarioWithNumeration, String prefix) {
+        String line = makeTabulaturePrefix(node.getNestingLevel()) + prefix + node.getLine() + "\n";
+        scenarioWithNumeration.addLast(line);
+        if (node instanceof KeyNode) {
+            if (((KeyNode) node).hasChildren()) {
+                for (int i = 0; i < ((KeyNode) node).getChildrenCount(); i++) {
+                    addLineWithNumeration(((KeyNode) node).getChildren().get(i), scenarioWithNumeration, prefix + (i + 1) + ".");
+                }
+            }
+        }
+    }
+
+    /**
+     * Return a scenario in base form
+     * @return Scenario as text
+     */
+    public String getScenario() {
+        LinkedList<String> scenario = new LinkedList<>();
+        for (Node firstLevelNode : getFirstLevelNodes()) {
+            if (firstLevelNode instanceof KeyNode) {
+                if (((KeyNode) firstLevelNode).hasChildren()) {
+                    addLine(firstLevelNode, scenario);
+                }
+            } else {
+                String line = makeTabulaturePrefix(firstLevelNode.getNestingLevel()) + firstLevelNode.getLine() + "\n";
+                scenario.addLast(line);
+            }
+
+        }
+        return changeLinkedListToString(scenario);
+    }
+
+    /**
+     * Add line to specific node
+     * @param node Node
+     * @param scenarioWithNumeration Scenario list
+     */
+    private void addLine(Node node, LinkedList<String> scenarioWithNumeration) {
+        String line = makeTabulaturePrefix(node.getNestingLevel()) + node.getLine() + "\n";
+        scenarioWithNumeration.addLast(line);
+        if (node instanceof KeyNode) {
+            if (((KeyNode) node).hasChildren()) {
```

Comment:
I don't like this double cast to KeyNode

Reply:
If i dont check node is KeyNode in second if Throw ClassCastException